This patch enables line number information for debugging assembler
source files.

--- gcc-3.3.2/gcc/config/mips/mips.h.orig	2003-05-30 08:00:42.000000000 -0400
+++ gcc-3.3.2/gcc/config/mips/mips.h	2004-01-31 15:51:24.000000000 -0500
@@ -962,11 +962,11 @@
 #endif
 
 /* GAS_ASM_SPEC is passed when using gas, rather than the MIPS
    assembler.  */
 
-#define GAS_ASM_SPEC "%{mtune=*} %{v}"
+#define GAS_ASM_SPEC "%{mtune=*} %{v} %{g:%{.S:-gstabs} %{.s:-gstabs}}"
 
 
 extern int mips_abi;
 
 #ifndef MIPS_ABI_DEFAULT
